src/category_theory/mwe.lean

```diff
@@ -0,0 +1,76 @@
+import data.fin
+import tactic
+
+def hom (n m : ℕ) := fin n → fin m
+def map {n m : ℕ} (f : hom n m) : hom (n+1) (m+1) :=
+λ i, if h : i.val < n then (f (i.cast_lt h)).cast_succ else fin.last _
+
+lemma map_increasing {n m : ℕ} (f: hom n m) (w : monotone f) : monotone (map f) :=
+λ a b h,
+begin
+  dsimp [map],
+  split_ifs,
+  {tidy},
+  {apply fin.le_last},
+  {rw [fin.le_iff_val_le_val] at h,
+  linarith},
+  {apply fin.le_last}
+end
+
+lemma fooo {a b n : ℕ} (h1 : a ≤ b) (h2 : ¬(a < n)) :¬ (b < n) := by library_search
+
+lemma map_id {n m : ℕ} (f : hom n m) : map (@id (fin n)) = @id (fin (n+1)) :=
+funext (λ a,
+begin
+  dsimp [map],
+  split_ifs,
+  {tidy},
+  {exact (fin.ext_iff _ _).2  (eq.trans rfl (eq.symm (nat.eq_of_lt_succ_of_not_lt a.is_lt h)))}
+end)
+
+
+lemma cast_succ_val_lt {n : ℕ} (i : fin n) : (fin.cast_succ i).val < n :=
+begin
+ rw [fin.cast_succ_val],
+ exact i.is_lt,
+end
+
+lemma cast_lt_cast_succ {n : ℕ} (i : fin n)  :
+  fin.cast_lt (fin.cast_succ i) (cast_succ_val_lt _) = i :=
+fin.eq_of_veq (by simp only [fin.cast_lt_val, fin.cast_succ_val])
+
+lemma map_comp {l m n : ℕ} (f : hom l m) (g : hom m n) : map (g ∘ f) = (map g) ∘ (map f) :=
```

Comment:
Thank for completing this solution. If you're interested, we could have a look at `split_ifs` sometime and see if we can see what was going wrong. It would be good to fix the underlying issue if possible. The fact that the second `split_ifs` call succeeds but doesn't do anything presumably means it knows what it's meant to be doing but for some reason can't execute it, and we should be able to step through `split_ifs` and work out exactly what goes wrong.

Reply:
For the time being I'd like to stay focused on this homotopy.io project but it would be interesting to see what changes could be made to `split_ifs` and `linarith`.